app/api/probe_dock/project_versions_api.rb

```diff
@@ -50,7 +50,7 @@ def current_organization
           end
 
           if params[:testId].present?
-            paginated_rel = paginated_rel.joins(test_results: :test).where('project_tests.api_id = ?', params[:testId])
+            paginated_rel = paginated_rel.joins(test_results: :test).where('project_tests.api_id = ?', params[:testId]).group('project_versions.id, projects.name')
```

Comment:
Usually when there's a relation that has to have a group clause depending on the parameters, I prefer to use a boolean `group` variable like this:

```
paginated rel do |rel|
  group = false

  if params[:a]
    group = true
    ...
  end

  if params[:b]
    ...
  end

  if params[:c]
    group = true
    ...
  end

  rel = rel.group(...) if group
end

It avoids repeating the group clause if multiple many-to-many joins make it necessary (since it's usually always the same group clause). Even if there's only one parameter that requires it at the moment, doing it like this discourages code duplication.